src/common/containers/build-details.js

```diff
@@ -38,12 +38,13 @@ class BuildDetails extends Component {
           snapcraft cleanbuild --debug
         </HelpInstallSnap>
       );
-    } else if (snap && snap.store_name) {
-      // otherwise if we have snap name show install instructions
+    } else if (snap && snap.store_name && build.storeRevision) {
```

Comment:
We don't need to support the case where we have snap name but not the revision?
In such case it means build failed (at least failed to publish), so it cannot be installed?